src/other/broadcasting.jl

```diff
@@ -121,6 +121,28 @@ end
 Base.dotview(df::SubDataFrame, ::typeof(!), idxs) =
     throw(ArgumentError("broadcasting with ! row selector is not allowed for SubDataFrame"))
 
+
+# TODO: remove the deprecations when Julia 1.7 functionality is commonly used
```

Comment:
@nalimilan - can you please check this. I will update tests and documentation once we decide what we like.

Reply:
bump (no rush of curse).

The key question is if when we write:

```
df.col .= value
```
and `col` is present in the data frame do we want this operation to always work (even for `SubDataFrame`) and be in-place - this is a current behavior

or

we want this to replace a column and be allowed only for `DataFrame`.

The reason I am asking is that when someone writes:
```
df.col .= value
```
maybe it is natural for broadcasting to work on the column `col` extracted from `df` (this is our current behavior)

and the only thing we should change is make `df.col .= value` work if `col` is not present (and allocate it then).

The tension is that if we follow the reasoning above `df.col .= value` is not consistent with `df[!, :col] .= value`. But maybe it is not a problem?

Reply:
Just to highlight the crucial tension in the decision to be made here. Currently we have:
```
julia> using DataFrames

julia> df = DataFrame(a=1:4)
4×1 DataFrame
 Row │ a     
     │ Int64 
─────┼───────
   1 │     1
   2 │     2
   3 │     3
   4 │     4

julia> df.a .= 'a':'d'
4-element Vector{Int64}:
  97
  98
  99
 100

julia> df
4×1 DataFrame
 Row │ a     
     │ Int64 
─────┼───────
   1 │    97
   2 │    98
   3 │    99
   4 │   100
```
and this is what happens if `:a` is present in `df`. If `:a` were not present in `df` then after this PR it would hold `['a','b','c','d']`. The question is if it is better to:
* keep an in-place operation we have now (with the inconsistent behaviors between existing and new column operations, but it is a corner case, and probably normally users might want an in-place behavior)
* deprecate in-place operation and in the future replace column (but this will then also mean that `df.col .= value` will not be allowed for `SubDataFrame`)

Both options have pros and cons. @pdeffebach - maybe you can comment from user's perspective which one would seem more convenient in daily work?.

Reply:
OK - then I will finalize this PR with deprecations injected into code. Thank you!